src/osp/Active/parts.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2022 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#pragma once
+
+#include "activetypes.h"
+
+#include "../Resource/resourcetypes.h"
+#include "../types.h"
+
+#include <longeron/id_management/registry.hpp>
+#include <longeron/containers/intarray_multimap.hpp>
+
+namespace osp::active
+{
+
+using PartEnt_t = uint32_t;
+using RigidGroup_t = uint32_t;
```

Comment:
Not using the strong-typedef of the enum classes for these?

No objection, just asking.

Reply:
I figured it isn't that helpful due to the excessive std::size_t(...) casts required. They're only useful when used as function parameters, and so far I don't have a good solution to generate a function that accepts an enum class just to access a single container. Prefixing the variable names with what type of index they accept decently helpful.